src/v2/System/useFeatureFlag.tsx

```diff
@@ -11,6 +11,16 @@ interface FeatureFlagDetails {
   variant: Variant
 }
 
+interface VariantTrackingProperties {
+  experiment_name: string
+  variant_name: string
+  payload?: string
+  context_owner_type: OwnerType
+  context_owner_id?: string
+  context_owner_slug?: string
+  should_track?: boolean
```

Comment:
not sure if the name is the best since we have a should_track function but with this way making it optional and defaulting to true we will be able to reuse this. @dblandin @mc-jones what do you think?

Reply:
I like that approach!

On a separate non-blocking note: it feels a bit odd to have snake_case keys in the interface here. If this is meant to be an abstracted API, would it make sense to support camelCase keys and map/convert before the internal trackEvent call?

Reply:
> On a separate non-blocking note: it feels a bit odd to have snake_case keys in the interface here. If this is meant to be an abstracted API, would it make sense to support camelCase keys and map/convert before the internal trackEvent call?

gave it a go I think it is ready 🤩 when you have some time just have another look 🎉